swagger/src/main/scala/org/http4s/rho/swagger/TypeBuilder.scala

```diff
@@ -80,23 +82,64 @@ object TypeBuilder {
 
           models ++ generics ++ children
 
+        case tpe@TypeRef(_, sym: Symbol, tpeArgs: List[Type]) if isSumType(sym) =>
+          // TODO promote methods on sealed trait from children to model
+          modelToSwagger(tpe, sfs).map(addDiscriminator(sym)).toSet.flatMap { model =>
+            val parent = RefModel(model.id + "Ref", model.id2 + "Ref", model.id2)
+            val children =
+              sym.asClass.knownDirectSubclasses.flatMap( sub =>
+                  modelToSwagger(sub.asType.toType, sfs).map(composedModel(parent))
+                )
+            alreadyKnown ++ Set(model) ++ children
+          }
+
         case e =>
           Set.empty
       }
 
     go(t, alreadyKnown, known)
   }
 
+  private def addDiscriminator(sym: Symbol)(model: ModelImpl): ModelImpl = {
+    val typeVar = sym.annotations
+      .withFilter( a => a.tpe <:< typeOf[DiscriminatorField] )
+      .flatMap(a => a.tree.children.tail.collect { case Literal(Constant(field: String)) => field } )
+      .headOption.getOrElse("type")
```

Comment:
`type` is the default for most libraries in terms of sum encoding. [argonaut-shapeless](https://github.com/alexarchambault/argonaut-shapeless#custom-encoding-for-sealed-traits), [spray-json-shapeless](https://github.com/fommil/spray-json-shapeless/blob/master/src/main/scala/fommil/sjs/FamilyFormats.scala#L394). Circe doesn't have a default.